tally_ho/apps/tally/views/data/candidate_list_view.py

```diff
@@ -33,10 +36,21 @@ def render_column(self, row, column):
 
     def filter_queryset(self, qs):
         tally_id = self.kwargs.get('tally_id', None)
+        office_id = self.kwargs.get('office_id', None)
+        keyword = self.request.GET.get('search[value]', None)
 
-        if tally_id:
+        if tally_id and office_id:
+            candidate_ids =\
+                get_office_candidates_ids(
+                    office_id=office_id, tally_id=tally_id)
+            qs = qs.filter(pk__in=candidate_ids)
+
+        if tally_id and not office_id:
             qs = qs.filter(tally__id=tally_id)
```

Comment:
Seems we aren't handling a situation when `tally_id = None` and `office_id != None`.  Is this intentional?

Reply:
Cool, so for `tally_id` it's intentional since this route always requires a `tally_id` but the `office_id` is optional, and for `office_id != None` since `None` always evaluates to `false` am checking for a truth value from `office_id` at line 42.